assets/js/script.js

```diff
@@ -2,80 +2,76 @@
 
-var generatePassword = function(){
+const generatePassword = function(){
 
   //Defining all possible characters for password and password length in object
-  var passwordChars = {
+  const passwordChars = {
     upperCase:"ABCDEFGHIJKLMNOPQRSTUVWXYZ",
     lowerCase:"abcdefghijklmnopqrstuvwxyz",
     numerics:"1234567890",
     specialChar:"!'`~@#$%^&*()_+-][}{|><?;:=",
     passwordLimit: [8,128]
   }
 
+  const SELECT_LENGTH_PROMPT = "Please choose your password length between 8-128 characters."
+  const INVALID_LENGTH_PROMPT = "Please choose a valid length for your password."
+  const UPPERCASE_CONFIRMATION_PROMPT= "Would you like Uppercase Characters in your password?\
+    \nClick Ok to add Uppercase Characters to your password.\
+    \nClick Cancel to not use Uppercase Characters in your password."
+  const LOWERCASE_CONFIRMATION_PROMPT = "Would you like Lowercase Characters in your password?\
+    \nClick Ok to add Lowercase Characters to your password.\
+    \nClick Cancel to not use Lowercase Characters in your password."
+  const NUMERIC_CONFIRMATION_PROMPT = "Would you like Numerical Characters in your password?\
+    \nClick Ok to add Numerical Characters to your password.\
+    \nClick Cancel to not use Numerical Characters in your password."
+  const SPECIAL_CHAR_CONFIRMATION_PROMPT = "Would you like Special characters in your password?\
+    \nClick Ok to add Special Characters to your password.\
+    \nClick Cancel to not use Special Characters in your password."
+  const NO_CHAR_TYPE_SELECTED_ERROR = "You need to pick at least 1 option to generate a password. Try again!"
+
   //Defining container for characters to be used in password
-  var pwdAllowableChar = "";
+  let pwdAllowableChar = "";
 
   //Prompt user to choose password length
-  var passwordLength = prompt("Please choose your password length between 8-128 characters.");
+  let passwordLength = prompt(SELECT_LENGTH_PROMPT);
  
   //Turn passwordLength into integer instead of string
   passwordLength = parseInt(passwordLength);
   console.log(passwordLength);
 
   //Check if user selected an acceptable password length
   if (passwordLength < passwordChars.passwordLimit[0] || passwordLength > passwordChars.passwordLimit[1] || isNaN(passwordLength) === true){
-    alert("Please choose a valid length for your password.");
+    alert(INVALID_LENGTH_PROMPT);
     return generatePassword();
   }
 
-  //Prompt user to choose to use uppercase characters or not
-  var useUpperCase = confirm("Would you like Uppercase Characters in your password?\
-  \nClick Ok to add Uppercase Characters to your password.\
-  \nClick Cancel to not use Uppercase Characters in your password.");
-
   //Add uppercase characters to list of characters to be used
-  if(useUpperCase === true){
-    pwdAllowableChar = pwdAllowableChar + passwordChars.upperCase;
+  if(confirm(UPPERCASE_CONFIRMATION_PROMPT)){
+    pwdAllowableChar += passwordChars.upperCase;
```

Comment:
Nifty way to add something to and existing "let" variable without changing the name.  You can also do "-="